packages/frontend/hooks/src/use-workspace-blob.ts

```diff
@@ -0,0 +1,37 @@
+import { workspaceManagerAtom } from '@affine/workspace/atom';
+import type { WorkspaceMetadata } from '@affine/workspace/metadata';
+import { useAtomValue } from 'jotai';
+import { useEffect, useState } from 'react';
+
+export function useWorkspaceBlobObjectUrl(
+  meta?: WorkspaceMetadata,
+  blobKey?: string | null
+) {
+  const workspaceManager = useAtomValue(workspaceManagerAtom);
+
+  const [blob, setBlob] = useState<string | undefined>(undefined);
+
+  useEffect(() => {
+    setBlob(undefined);
+    if (!blobKey || !meta) {
+      return;
+    }
+    let canceled = false;
+    workspaceManager
+      .getWorkspaceBlob(meta, blobKey)
+      .then(blob => {
+        if (blob && !canceled) {
+          setBlob(URL.createObjectURL(blob));
```

Comment:
1. better revoke the object url if it is not used any longer
2. or, we can cache the url if it is already in cache